pkg/shp/cmd/buildrun/create.go

```diff
@@ -60,6 +62,13 @@ func (c *CreateCommand) Run(params *params.Params, ioStreams *genericclioptions.
 		},
 		Spec: *c.buildRunSpec,
 	}
+
+	envs, err := c.cmd.Flags().GetStringArray("env")
+	if err != nil {
+		return err
+	}
+	br.Spec.Env = append(br.Spec.Env, util.StringSliceToEnvVarSlice(envs)...)
```

Comment:
This type of action should be done during `Complete()` (or maybe `Prepare()`), so the `Run` method can concentrate on executing the task.

Reply:
The BuildRun object does not exist until the `Run` function ....

Reply:
Maybe not as a subject for this pull-request, but being it the `buildrun` sub-command, I think it would make sense to deal with a `BuildRun` object instead of a `Build`... thinking out loud.

Reply:
as I recall env vars in the shipwright-io/build PR and associated SHIP EP can be set on both the Build and BuildRun

reasonable defaults are set on the Build

overrides on atypical adds (i.e. not set on the Build) are set on the BuildRun

discussion on the pros and cons of this, if you want to continue @otaviof , as you suggested, should span beyond this PR

given where the EP and build PR is, I would say we move forward with this as is, and if you want to continue to pursue this @otaviof , we start with a community discussion in the Monday meeting, and see where that goes from there

Reply:
Sure! Make sense, @gabemontero!

Reply:
cool thanks !